editor/src/components/editor/store/dispatch.tsx

```diff
@@ -646,6 +652,24 @@ export function editorDispatchClosingOut(
       updatedFromVSCodeOrParsedAfterCodeChange,
     )
     applyProjectChangesToVSCode(frozenEditorState, projectChanges)
+    if (finalStore.collaborativeEditingSupport.session != null) {
```

Comment:
couldn't this block move above the creation of the original `finalStore`? so by the time we make the finalStore we have the final_final filesModifiedByAnotherUser value?

Reply:
Unless there's a strong case for making this change now, I'd rather leave this for a subsequent refactor.